### PR TITLE
Add flair for users who donated to the donation election

### PR DESCRIPTION
**Pull request by Will-Howard**
This adds an icon next to a user's name (on posts and comments) that looks like this:
Screenshot 2023-11-17 at 17 58 07

To enable this an admin has to manually set a flag on the user (as we will be requiring people to send us their donation receipts to add this)

@agnesstenlund Feel free to suggest a better icon. Another question is whether we should show this on user profiles in addition to on posts and comments (as a lot of users never comment)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205979924938947) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/users/UserCommentMarkers.tsx [hunk 1]
```diff
@@ -38,8 +44,9 @@ const UserCommentMarkers = ({
 
   const showAuthorIcon = isFriendlyUI && isPostAuthor;
   const showNewUserIcon = isNewUser(user);
+  const _showDonatedIcon = showDonatedIcon(user);
 
-  if (!showAuthorIcon && !showNewUserIcon) {
+  if (!showAuthorIcon && !showNewUserIcon && !showDonatedIcon) {
```
Comment (jpaddison3) [inline packages/lesswrong/components/users/UserCommentMarkers.tsx:49]:
```suggestion
  if (!showAuthorIcon && !showNewUserIcon && !_showDonatedIcon) {
```

File: packages/lesswrong/lib/collections/users/schema.ts [hunk 2]
```diff
@@ -2830,7 +2830,20 @@ const schema: SchemaType<DbUser> = {
     canRead: ['guests'],
     hidden: true,
     optional: true
-  }
+  },
+
+  // Giving season fields
+  givingSeason2023DonatedFlair: {
+    type: Boolean,
```
Comment (jpaddison3) [inline packages/lesswrong/lib/collections/users/schema.ts:2837]: Are we gonna add another flair for 'voted'? IMO make this an enum.

**Review by jpaddison3 — APPROVED** [review]
I propose you go with a different approach. Feel free to go with this approach if you disagree with me.

File: packages/lesswrong/components/users/UserCommentMarkers.tsx (hunk 1 above)
Comment (Will-Howard) [reply packages/lesswrong/components/users/UserCommentMarkers.tsx:49]: Good catch!

File: packages/lesswrong/lib/collections/users/schema.ts (hunk 2 above)
Comment (Will-Howard) [reply packages/lesswrong/lib/collections/users/schema.ts:2837]: We are, but I would prefer to keep it as a boolean for simplicity + consistency with the others. Also we may in future want to see independently see who voted and who (we know) donated

**Comment by Will-Howard** [comment]
This is what it looks like on the profile page:
Screenshot 2023-11-21 at 12 58 03

> I propose you go with a different approach. Feel free to go with this approach if you disagree with me.

Assuming you're talking about the boolean vs enum thing, I'm going to stick with boolean for the reasons given above

**Pull request merged**